app/src/main/java/com/example/ohthmhyh/activities/UpdateHabitEventActivity.java

```diff
@@ -102,6 +114,38 @@ public boolean onOptionsItemSelected(@NonNull MenuItem item) {
         }
         return super.onOptionsItemSelected(item);
     }
+    /**
+     * This handles the drop down list for Habits
+     * Can tell if we are on edit or new Calls to the data base to get the habitevent list
+     */
+    private void makeNameList(){
+        ArrayList<String> NameList=new ArrayList<>();
+        NameList.addAll(habitList.getHabitNames());
```

Comment:
Stu, we are using lower camel case naming (https://en.wikipedia.org/wiki/Camel_case). I also suggest renaming this to something more specific:
```suggestion
        ArrayList<String> habitNameList = habitList.getHabitNames();
```
Note that if you accept these changes, you will have to go through and also rename all the `NameList` references to `habitNameList`.

Reply:
Sounds good